### PR TITLE
Modernise steinberger

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR now uses `MaterialUtilities::compute_only_composition_fractions` in steinberger.

### CODE REVIEW

File: source/material_model/steinberger.cc [hunk 1]
```diff
@@ -495,50 +487,49 @@ namespace aspect
...
-                                                                                        n_conductivity_layers,
-                                                                                        "Reference temperatures for thermal conductivity");
-          conductivity_exponents = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Thermal conductivity exponents"))),
-                                                                           n_conductivity_layers,
-                                                                           "Thermal conductivity exponents");
-          saturation_scaling = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Saturation prefactors"))),
-                                                                       n_conductivity_layers,
-                                                                       "Saturation prefactors");
```
Comment (gassmoeller) [inline source/material_model/steinberger.cc:516]: I do not quite see the benefit of changing this from a list to parsing a map. Is there any use case we do not support with the current version or something we can do cleaner with the map parsing? I dont think we have to convert all input parameter parsing to the map, unless there is a benefit to it.

**Review by gassmoeller — COMMENTED** [review]
Just one question. Thanks for the cleanup!

File: source/material_model/steinberger.cc (hunk 1 above)
Comment (bobmyhill) [reply source/material_model/steinberger.cc:516]: The only benefit at the moment is that (I think) it looks a bit cleaner. Maybe in the future someone will want to copy this into another model that might need multiple phases?

Comment (bobmyhill) [reply source/material_model/steinberger.cc]: I've restored the original format, which reduces the line changes.

**Pull request merged**